File: app/src/main/java/com/android/logoped/SectionsPagerAdapter.java
package com.android.logoped;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.android.logoped.WordImageFragment;
import com.android.logoped.model.Word;

import java.util.List;

/**
 * Created by paul on 18/02/15.
 */
public class SectionsPagerAdapter extends FragmentStatePagerAdapter {
    private List<Word> mWords;

    public SectionsPagerAdapter(FragmentManager fm, List<Word> words) {
        super(fm);
        mWords = words;
    }

    @Override
    public Fragment getItem(int position) {
        return WordImageFragment.newInstance(mWords.get(position));
    }

    @Override
    public int getCount() {
        return mWords.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mWords.get(position).getName();
    }
}

File: app/src/main/java/com/pps/ppls/SectionsPagerAdapter.kt
package com.pps.ppls

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import com.pps.ppls.database.DbManager
import com.pps.ppls.model.Word
import java.util.concurrent.CopyOnWriteArrayList

/**
 * Created by paul on 18/02/15.
 */
class SectionsPagerAdapter(fragmentManager: FragmentManager, val context: Context) : FragmentStatePagerAdapter(fragmentManager) {
    private val words = CopyOnWriteArrayList<Word>()

    init {
        words.addAll(DbManager.getInstance(context).allWords)
        words.shuffle()
    }

    override fun getItem(position: Int): Fragment {
        return WordImageFragment.newInstance(words[position])
    }

    override fun getCount(): Int {
        return words.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return words[position].name
    }

    fun getWordAt(index: Int): Word {
        return words[index]
    }
}